src/drivers/imu/adis16477/ADIS16477.cpp

```diff
@@ -545,109 +320,18 @@ ADIS16477::measure()
-	sensor_accel_s arb = {};
-	arb.timestamp = hrt_absolute_time();
-	arb.device_id = _device_id.devid;
-	arb.error_count = perf_event_count(_bad_transfers);
-
-	// raw sensor readings
-	arb.x_raw = report.accel_x;
-	arb.y_raw = report.accel_y;
-	arb.z_raw = report.accel_z;
-	arb.scaling = _accel_range_scale;
-
-	float xraw_f = report.accel_x * _accel_range_scale;
-	float yraw_f = report.accel_y * _accel_range_scale;
-	float zraw_f = report.accel_z * _accel_range_scale;
-
-	// apply user specified rotation
-	rotate_3f(_rotation, xraw_f, yraw_f, zraw_f);
-
-	float x_in_new = (xraw_f - _accel_scale.x_offset) * _accel_scale.x_scale;
-	float y_in_new = (yraw_f - _accel_scale.y_offset) * _accel_scale.y_scale;
-	float z_in_new = (zraw_f - _accel_scale.z_offset) * _accel_scale.z_scale;
-
-	arb.x = _accel_filter_x.apply(x_in_new);
-	arb.y = _accel_filter_y.apply(y_in_new);
-	arb.z = _accel_filter_z.apply(z_in_new);
-
-	matrix::Vector3f aval(x_in_new, y_in_new, z_in_new);
-	matrix::Vector3f aval_integrated;
-
-	bool accel_notify = _accel_int.put(arb.timestamp, aval, aval_integrated, arb.integral_dt);
-	arb.x_integral = aval_integrated(0);
-	arb.y_integral = aval_integrated(1);
-	arb.z_integral = aval_integrated(2);
-
-	/* Temperature report: */
-	// temperature 1 LSB = 0.1°C
-	arb.temperature = report.temp * 0.1;
-
-	if (accel_notify) {
-		orb_publish(ORB_ID(sensor_accel), _accel_topic, &arb);
-	}
-
-	return accel_notify;
-}
-
-bool
-ADIS16477::publish_gyro(const ADISReport &report)
-{
-	sensor_gyro_s grb = {};
-	grb.timestamp = hrt_absolute_time();
-	grb.device_id = _gyro->_device_id.devid;
-	grb.error_count = perf_event_count(_bad_transfers);
-
-	/* Gyro report: */
-	grb.scaling = math::radians(_gyro_range_scale);
-	grb.x_raw = report.gyro_x;
-	grb.y_raw = report.gyro_y;
-	grb.z_raw = report.gyro_z;
-
-	// ADIS16477-2BMLZ scale factory
-	float xraw_f = report.gyro_x;
-	float yraw_f = report.gyro_y;
-	float zraw_f = report.gyro_z;
-
-	// apply user specified rotation
-	rotate_3f(_rotation, xraw_f, yraw_f, zraw_f);
-
-	float x_gyro_in_new = (math::radians(xraw_f * _gyro_range_scale) - _gyro_scale.x_offset) * _gyro_scale.x_scale;
-	float y_gyro_in_new = (math::radians(yraw_f * _gyro_range_scale) - _gyro_scale.y_offset) * _gyro_scale.y_scale;
-	float z_gyro_in_new = (math::radians(zraw_f * _gyro_range_scale) - _gyro_scale.z_offset) * _gyro_scale.z_scale;
-
-	grb.x = _gyro_filter_x.apply(x_gyro_in_new);
-	grb.y = _gyro_filter_y.apply(y_gyro_in_new);
-	grb.z = _gyro_filter_z.apply(z_gyro_in_new);
-
-	matrix::Vector3f gval(x_gyro_in_new, y_gyro_in_new, z_gyro_in_new);
-	matrix::Vector3f gval_integrated;
-
-	bool gyro_notify = _gyro_int.put(grb.timestamp, gval, gval_integrated, grb.integral_dt);
-	grb.x_integral = gval_integrated(0);
-	grb.y_integral = gval_integrated(1);
-	grb.z_integral = gval_integrated(2);
-
 	/* Temperature report: */
 	// temperature 1 LSB = 0.1°C
-	grb.temperature = report.temp * 0.1f;
+	const float temperature = adis_report.temp * 0.1f;
```

Comment:
I'd love to see a variable or #define replace the magic number here... see PR #11629